llvm-project/stage2.cmake

```diff
@@ -0,0 +1,22 @@
+# TODO: comment every one of these.
```

Comment:
Just an FYI, there can probably be some more refactoring done on these caches as they grow so that more values can be shared:

https://github.com/llvm/llvm-project/blob/1e14b1a7977b0bdc725312f76ebe7173f6f1f62c/clang/cmake/caches/3-stage.cmake#L16

At this point, I don't think it matters but it might make maintenance over the long term a little easier.

Reply:
Neat! One thing I dislike a bit about the AOSP llvm build system is that I feel that aggressive factorization of cmake variables makes it hard to just have the full list at any given time to see precisely what's getting invoked.

That said, some of the variables like "please default to compiler-rt" are important enough that they probably should be supplied to every cmake invocation. Consolidating them is probably less error prone, since it makes it less difficult to accidentally forget to set a certain value.

Reply:
Yes, I think there is a balance with that but certain things like defaulting to `Release` and defaulting certain things on or off across all stages should be moved out (maybe `base.cmake`?).

Reply:
Makes sense. Let's get more of the pipeline filled in, then start seeing where refactorings make sense.